core-bundle/src/Routing/Content/ContentUrlResolverInterface.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Routing\Content;
+
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Route;
+
+/**
+ * A content URL resolver converts content to a URL (represented by a route object).
+ *
+ * There are multiple valid results for a resolver:
+ *
+ * 1. Resolve to a Contao\CoreBundle\Routing\Content\PageRoute if the
+ *    content is embedded on a page (e.g. through a "reader" module).
+ *
+ * 2. Resolve to a Symfony\Component\Routing\Route if the content's URL
+ *    is not within the CMS (e.g. an absolute URL).
+ *
+ * 3. Resolve to a different content that represents this content
+ *    (e.g. if the content has no "full" representation of its own).
+ *
+ * If a resolver is responsible for a content object but cannot convert it to a route,
+ * a Symfony\Component\Routing\Exception\RouteNotFoundException should be thrown.
+ */
+interface ContentUrlResolverInterface
+{
+    public const ATTRIBUTE = '_content_resolver';
+
+    /**
+     * Returns a Symfony\Component\Routing\Route or another $content to resolve.
+     *
+     * @param mixed $content The route "content" which may be an object or anything
+     *
+     * @return Route|PageRoute|mixed
```

Comment:
This is code smell. It can return a route, a page route or whatever. Having reviewed that PR for quite some time now, I still don't understand what I need to return here. Not even by reading the three examples.

Can you explain why my content cannot just return the correct `Route`?

Reply:
Because a content can resolve to another content. A news can point to a page, to an article or to an external page. If it points to an article, the news resolver shouldn't be responsible to resolve the article URL. It simply returns an article, which means all resolvers will be asked again in turn to resolve the new content.

Reply:
Changed in 1fc7bfd603daf90caa6d9515664f2ac98361c760. The resolver now always returns a route, so it's responsible for _forwarding_ a different content type.